### ISSUE TITLE
Example for implementing idle session timeout

### ISSUE THREAD

**Issue by omerkirk** [olahol/melody#70]
Hi, 

Thank you for a great package, it is much cleaner and simpler to use compared to gorilla/websocket library. 

I am developing a game server and need to implement an idle timeout, I've set PingPeriod and PongWait values however it doesn't timeout the session or close the connection. In the readme feature list it looks like this is included "Automatic handling of ping/pong and session timeouts" however I can't seem to get it to work. Can you add an example that includes a simple implementation.

**Comment by olahol** [comment]
The doc is a little confusing but the automatic ping/pong handling is for [Websocket heartbeats](https://developer.mozilla.org/en-US/docs/Web/API/WebSockets_API/Writing_WebSocket_servers#pings_and_pongs_the_heartbeat_of_websockets) which time out broken connection (i.e sessions that don't respond to the control frame 'ping').

**Comment by kokteyldev** [comment]
That is right thank you, we are doing some cleanup on HandleDisconnect for each session and we realized there are sometimes sessions that are no longer available but the cleanup process didn't run for some reason. We are looking into it, our current understanding is in some cases HandleError runs but disconnecthandler is not called. I will update this issue when I complete debugging.

**Comment by youngsailor** [comment]
Has this bug been fixed ？

**Comment by youngsailor** [comment]
> 



> That is right thank you, we are doing some cleanup on HandleDisconnect for each session and we realized there are sometimes sessions that are no longer available but the cleanup process didn't run for some reason. We are looking into it, our current understanding is in some cases HandleError runs but disconnecthandler is not called. I will update this issue when I complete debugging.



> 

I have the same problem. Have you solved your problem?

**Comment by olahol** [comment]
In `v1.1.2` `HandleDisconnect` should now run more reliably.
